### PR TITLE
Fix ReplaceTokens bug in AttributeRouteModel

### PR DESCRIPTION
**Pull request by AlexejTimonin**
Add scope to track when the token ends.

Addresses #6105

### CODE REVIEW

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs [hunk 1]
```diff
@@ -231,6 +231,7 @@ public static string ReplaceTokens(string template, IDictionary<string, string>
             var state = TemplateParserState.Plaintext;
 
             int? tokenStart = null;
+            int scope = 0;
```
Comment (JamesNK) [inline src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs]: nit: var

(it's ASP.NET team coding standard)

**Review by JamesNK — CHANGES_REQUESTED** [review]
Looks good!

I think we should also test for unbalanced nested brackets. Could you add some? For example:
* `"[controller]/[[[action]]/{id}"`
* `"[controller]/[[[action]]]]/{id}"`
* `"[controller]/[[action]]]/{id}"`

I would expect all those situations to throw.

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs (hunk 1 above)
Comment (AlexejTimonin) [reply src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs]: Thank you for the review, I'll add more tests and change to var. Just noticed I've a redundant scope decrement. Sorry for my brain fart..

**Review by JamesNK — APPROVED** [review]

**Comment by JamesNK** [comment]
Thanks for fixing this. It will be in 3.0 preview3.

**Comment by AlexejTimonin** [comment]
Was fun! Thanks for guidance!

**Pull request merged**